engine/execution/ingestion/engine.go

```diff
@@ -876,62 +911,62 @@ func (e *Engine) handleCollection(originID flow.Identifier, collection *flow.Col
 		return fmt.Errorf("cannot store collection: %w", err)
 	}
 
-	return e.addCollectionToMempool(collection)
+	return e.mempool.BlockByCollection.Run(
+		func(backdata *stdmap.BlockByCollectionBackdata) error {
+			return e.addCollectionToMempool(collection, backdata)
+		},
+	)
 }
 
-func (e *Engine) addCollectionToMempool(collection *flow.Collection) error {
+func (e *Engine) addCollectionToMempool(collection *flow.Collection, backdata *stdmap.BlockByCollectionBackdata) error {
```

Comment:
Letting this function take the mempool data and moving the lock acquiring logic out of this function, so that it can also be used by `reloadBlock` function